tiledb/CMakeLists.txt

```diff
@@ -405,6 +405,11 @@ endif()
 # Compile options/definitions
 ############################################################
 
+if (NOT MSVC)
+# Enable exhaustive enum check even with default case
+  target_compile_options(TILEDB_CORE_OBJECTS PRIVATE -Wswitch-enum)
```

Comment:
I don't like the way that the warning configuration is not enabled for all platforms. Seems like it's asking for future complications. If we're turning it on for one compiler, we should endeavor to turn it on for all, and if it's not possible for some platform, that should be documented.